src/mcse.jl

```diff
@@ -133,6 +133,10 @@ function _mcse_sbm(f, x, batch_size)
     any(x -> x === missing, x) && return missing
     n = length(x)
     i1 = firstindex(x)
+    if all(==(first(x)), x)
```

Comment:
```suggestion
    if allequal(x)
```

Reply:
Since `allequal` was introduced in Julia v1.8, we would need to either drop support for Julia v1.6-1.7 or add Compat to use it.

Reply:
I think we should use `allequal` at least on newer Julia versions since there's no reason to avoid the specialized implementations for ranges or other special types. I assume if we don't use `allequal` right away, then it's very likely we'll never switch to the better/native approach even when support for older Julia versions is dropped.

I'd be fine with dropping support for 1.6 and 1.7 now that 1.10 is the new LTS (I'd even be fine with dropping support for < 1.10). Alternatively, to avoid adding a Compat dependency, we could also add our own custom, less-optimized function
```julia
if VERSION < v"1.8"
    _allequal(x) = isempty(x) || all(isequal(first(x)), x)
else
    const _allequal = allequal
end
```

Reply:
I'm okay with bumping support to v1.8. I see no problem with supporting v1.8 and v1.9 until it becomes at all cumbersome to do so.